products/batch_exports/backend/temporal/destinations/snowflake_batch_export.py

```diff
@@ -611,10 +620,29 @@ async def copy_loaded_files_to_snowflake_table(
             PURGE = TRUE
             """
 
+        # Handle cases where the Warehouse is suspended (sometimes we can recover from this)
+        max_attempts = 3
+        execute_copy_into = make_retryable_with_exponential_backoff(
+            self.execute_async_query,
+            max_attempts=max_attempts,
+            retryable_exceptions=(snowflake.connector.errors.ProgrammingError,),
+            # 608 = Warehouse suspended error
+            is_exception_retryable=lambda e: isinstance(e, snowflake.connector.errors.ProgrammingError)
+            and e.errno == 608,
+        )
+
         # We need to explicitly catch the exception here because otherwise it seems to be swallowed
         try:
-            result = await self.execute_async_query(query, poll_interval=1.0)
+            result = await execute_copy_into(query, poll_interval=1.0)
         except snowflake.connector.errors.ProgrammingError as e:
+            if e.errno == 608:
+                err_msg = (
+                    f"Failed to execute COPY INTO query after {max_attempts} attempts due to warehouse being suspended"
+                )
+                if e.msg is not None:
+                    err_msg += f": {e.msg}"
+                raise SnowflakeWarehouseSuspendedError(err_msg)
+
             self.logger.exception(f"Error executing COPY INTO query: {e}")
```

Comment:
Probably should put the log call above the new if block, so that we still log even when we raise a `SnowflakeWarehouseSuspendedError`, just in case.